tests/neg-scalajs/abstract-local-js-class.check

```diff
@@ -0,0 +1,4 @@
+-- Error: tests/neg-scalajs/abstract-local-js-class.scala:6:19 ---------------------------------------------------------
+6 |    abstract class AbstractLocalJSClass extends js.Object // error
+  |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  |    Implementation restriction: local JS classes cannot be abstract
```

Comment:
Because of the way we implement the local captures of the constructor. We need `ExplicitJSClasses` to create a fake `New(TheLocalJSClass, fakeParams)`, which lambdaLift (or its equivalent) enhances with the captures for local vals. The back-end captures the transformed fake `New` to recover those.

If the class is abstract, we cannot create that `New` without triggering internal inconsistencies. We also cannot blindly remove the `Abstract` flag, because the class could contain abstract methods, which still creates inconsistencies.